src/Methods/PaymentAbstract.php

```diff
@@ -245,7 +248,31 @@ public function isSwitchableTo($orderId = null): bool
      */
     public function isSwitchableFrom($orderId = null): bool
     {
-        // TODO ADD AMAZON PAY
-        return true;
+        if ($orderId > 0 ) {
+            /** @var OrderRepositoryContract $orderRepo */
+            $orderRepo = pluginApp(OrderRepositoryContract::class);
+            $filters = $orderRepo->getFilters();
+            $filters['addOrderItems'] = false;
```

Comment:
Why do you set all these filters?
It's not necessary here. You should use the lazy loading instead. (See next comment)